src/components/ReferenceSolutions/ReferenceSolutionDetail/ReferenceSolutionDetail.js

```diff
@@ -1,98 +1,166 @@
-            &nbsp;
-            <FormattedTime value={createdAt * 1000} />
-          </td>
-        </tr>
-        <tr>
-          <td className="text-center">
-            <FontAwesomeIcon icon="user" />
-          </td>
-          <th>
-            <FormattedMessage id="generic.author" defaultMessage="Author" />:
-          </th>
-          <td>
-            <UsersNameContainer userId={userId} />
-          </td>
-        </tr>
-      </tbody>
-    </Table>
-  </Box>;
+  }
+
+  render() {
+    const {
+      submission: {
+        id,
+        description,
+        runtimeEnvironmentId,
+        solution: { createdAt, userId, files }
+      },
+      exercise,
+      evaluations
+    } = this.props;
+    const { openFileId, activeSubmissionId } = this.state;
+
+    if (activeSubmissionId) {
+      var {
+        submittedBy,
+        evaluation,
+        isCorrect,
+        evaluationStatus,
+        ...restSub
+      } = evaluations.toJS()[activeSubmissionId].data;
+    } else evaluationStatus = 'missing-submission';
+    return (
+      <div>
+        <Row>
+          <Col md={6} sm={12}>
+            <ReferenceSolutionStatus
+              description={description}
+              solution={{ createdAt, userId }}
+              exerciseId={exercise.id}
+            />
+            <Row>
+              {files.map(file =>
+                <Col lg={6} md={12} key={file.id}>
+                  <a
+                    href="#"
+                    onClick={e => {
+                      e.preventDefault();
+                      this.openFile(file.id);
+                    }}
+                  >
+                    <SourceCodeInfoBox {...file} />
+                  </a>
+                </Col>
+              )}
+              {files.length > 1 &&
+                <Col lg={6} md={12}>
+                  <DownloadSolutionArchiveContainer
+                    solutionId={id}
+                    isReference={true}
+                  />
+                </Col>}
+            </Row>
+            {evaluation &&
+              <CompilationLogs
+                initiationOutputs={evaluation.initiationOutputs}
+              />}
+            <CommentThreadContainer threadId={id} />
+          </Col>
+
+          {evaluations &&
+            <Col md={6} sm={12}>
+              {evaluation &&
+                <TestResults
+                  evaluation={evaluation}
+                  runtimeEnvironmentId={runtimeEnvironmentId}
+                  isSupervisor={true}
+                />}
+
+              {evaluation &&
+                <Row>
+                  <Col lg={6} md={12}>
+                    <DownloadResultArchiveContainer
+                      submissionId={restSub.id}
+                      isReference={true}
+                    />
+                  </Col>
+                </Row>}
+              {activeSubmissionId &&
+                <Row>
+                  <Col lg={12}>
+                    <ResourceRenderer resource={evaluations.toArray()}>
```

Comment:
There is a better way, how ResourceRenderer can handle a single array of resources.